x-pack/plugins/synthetics/server/rest_api/synthetics_service/add_monitor_push.ts

```diff
@@ -0,0 +1,370 @@
+        reason: message,
+        details,
+        payload,
+      });
+      if (staleMonitorsMap[monitor.id]) {
+        staleMonitorsMap[monitor.id].stale = false;
+      }
+      return null;
+    }
+
+    const previousMonitor = await getExistingMonitor(savedObjectsClient, monitor.id, projectId);
+
+    if (previousMonitor) {
+      await updateMonitor(
+        savedObjectsClient,
+        encryptedSavedObjectsClient,
+        previousMonitor,
+        normalizedMonitor,
+        server
+      );
+      updatedMonitors.push(monitor.id);
+      if (staleMonitorsMap[monitor.id]) {
+        staleMonitorsMap[monitor.id].stale = false;
+      }
+    } else {
+      const newMonitor = await savedObjectsClient.create<EncryptedSyntheticsMonitor>(
+        syntheticsMonitorType,
+        formatSecrets({
+          ...normalizedMonitor,
+          revision: 1,
+        })
+      );
+      await syncNewMonitor({ server, monitor: normalizedMonitor, monitorSavedObject: newMonitor });
+      createdMonitors.push(monitor.id);
+    }
+  } catch (e) {
+    server.logger.error(e);
+    failedMonitors.push({
+      id: monitor.id,
+      reason: 'Failed to create or update monitor',
+      details: e.message,
+      payload: monitor,
+    });
+  }
+};
+
+const handleStaleMonitors = async ({
+  savedObjectsClient,
+  server,
+  staleMonitorsMap,
+  staleMonitors,
+  deletedMonitors,
+  keepStale,
+  failedStaleMonitors,
+}: {
+  savedObjectsClient: SavedObjectsClientContract;
+  server: UptimeServerSetup;
+  staleMonitorsMap: StaleMonitorMap;
+  staleMonitors: string[];
+  deletedMonitors: string[];
+  keepStale: boolean;
+  failedStaleMonitors: FailedMonitors;
+}) => {
+  try {
+    const staleMonitorsData = Object.values(staleMonitorsMap).filter(
+      (monitor) => monitor.stale === true
+    );
+    await Promise.all(
+      staleMonitorsData.map((monitor) => {
+        if (!keepStale) {
+          return deleteStaleMonitor({
+            deletedMonitors,
+            savedObjectsClient,
+            server,
+            monitorId: monitor.savedObjectId,
+            journeyId: monitor.journeyId,
+            failedStaleMonitors,
+          });
+        } else {
+          staleMonitors.push(monitor.journeyId);
+          return null;
+        }
+      })
+    );
+  } catch (e) {
+    server.logger.error(e);
+  }
+};
+
+const deleteStaleMonitor = async ({
+  deletedMonitors,
+  savedObjectsClient,
+  server,
+  monitorId,
+  journeyId,
+  failedStaleMonitors,
+}: {
+  deletedMonitors: string[];
+  failedStaleMonitors: FailedMonitors;
+  savedObjectsClient: SavedObjectsClientContract;
+  server: UptimeServerSetup;
+  monitorId: string;
+  journeyId: string;
+}) => {
+  try {
+    await deleteMonitor({ savedObjectsClient, server, monitorId });
+    deletedMonitors.push(journeyId);
+  } catch (e) {
+    failedStaleMonitors.push({
+      id: monitorId,
+      reason: 'Failed to delete stale monitor',
+      details: e.message,
+    });
+  }
+};
```

Comment:
i will highly recommend organising all these functions in a class , that way you will not have to pass objects like savedObjectsClient as parameter to each function. 

I envision something like this 
```

const pushMonitorsService  = new PushMonitorsService()
pushMonitorsService.initi(..thingsRequired)

... do rest of stuff
```
I think code will be lot cleaner and easier to maintain.